### PR TITLE
Add unit tests for Tera Blast to reproduce issue #9381.

### PR DESCRIPTION
**Pull request by vingkan**
Reproduces #9381 in a unit test.

If this change looks good, I can open a separate pull request to investigate a fix for the issue.

### CODE REVIEW

**Comment by vingkan** [comment]
> It you’re writing a test for behavior the sim gets wrong, you should write the test for what the behavior should be and `skip` the test since it fails.

Updated!

File: test/sim/moves/terablast.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		const battle = common.createBattle([[
+			// Regidrago has equal base attack and special attack stats.
+			{species: 'regidrago', ability: 'dragonsmaw', moves: ['terablast']},
+		], [
+			{species: 'regirock', ability: 'clearbody', moves: ['protect']},
+		]]);
+		battle.makeChoices('move terablast terastallize', 'move protect');
+
+		const leftPokemon = battle.p1.pokemon[0];
+		assert.notEqual(leftPokemon.terastallized, undefined);
```
Comment (DaWoblefet) [inline test/sim/moves/terablast.js:17]: It's not really necessary to check whether or not the Pokemon is Terastallized, that's pretty trivial. I would remove these checks.

**Review by DaWoblefet — CHANGES_REQUESTED** [review]
Very good overall! Only one minor nitpick, everything else is perfectly fine.

File: test/sim/moves/terablast.js (hunk 1 above)
Comment (vingkan) [reply test/sim/moves/terablast.js]: Good point. Removed!

**Comment by DaWoblefet** [comment]
Thanks vingkan!

**Pull request merged**